### PR TITLE
[FLINK-29859][e2e] Running TPC-DS with adaptive batch scheduler supports custom exceptions check

### PR DESCRIPTION
**Pull request by wanglijie95**
## What is the purpose of the change
Speculative execution will cause some extra exceptions during execution, which are expected, we should skip these exceptions in logs check. The skipped exceptions are:
`ExecutionGraphException: The execution attempt`
`Cannot find task to fail for execution`
`ExceptionInChainedOperatorException: Could not forward element to next operator`
`CancelTaskException: Buffer pool has already been destroyed`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? (**not applicable**)

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/test-runner-common.sh [hunk 1]
```diff
@@ -58,23 +59,28 @@ function run_test {
...
+    local check_exceptions_action="$3"
 
     local time_elapsed=$(end_timer)
 
-    if [[ "${skip_check_exceptions}" != "skip_check_exceptions" ]]; then
+    if [[ "${check_exceptions_action}" == "skip_check_exceptions" ]]; then
+        echo "Checking of logs skipped."
+    elif [[ "${check_exceptions_action}" == "custom_check_exceptions" ]]; then
+        echo "Custom exception checking of logs."
+        local check_command="$4"
```
Comment (zhuzhurk) [inline flink-end-to-end-tests/test-scripts/test-runner-common.sh:77]: It's better to define it at the beginning of the method to clearly show all the params.

File: flink-end-to-end-tests/test-scripts/test_tpcds.sh [hunk 2]
```diff
@@ -24,86 +24,153 @@ USE_TABLE_STATS=true
...
+    # Clean-up generated data folder
+    ################################################################################
+
+    rm -rf "${TPCDS_DATA_DIR}"
+    echo "Deleted all files under $TPCDS_DATA_DIR"
+}
+
+function check_logs_for_exceptions_for_adaptive_batch_scheduler {
+    echo "Checking for exceptions for adaptive batch scheduler..."
+    exception_count=$(grep -rv "GroupCoordinatorNotAvailableException" $FLINK_LOG_DIR \
```
Comment (zhuzhurk) [inline flink-end-to-end-tests/test-scripts/test_tpcds.sh:115]: It is possible to re-use the whitelisted exceptions in `check_logs_for_exceptions`, in `common.sh`?

**Review by zhuzhurk — COMMENTED** [review]
Thanks for creating this PR. @wanglijie95 
I have a few comments. PTAL.

**Comment by wanglijie95** [comment]
@zhuzhurk Thanks for review @zhuzhurk. I 've addressed the comments, please take a look.

**Review by zhuzhurk — APPROVED** [review]
LGTM.

**Pull request closed without merging**